config/config_test.go

```diff
@@ -373,6 +405,163 @@ func TestCheckMetricsEnabled(t *testing.T) {
 	}
 }
 
+func TestEnabledFlagGetsModified(t *testing.T) {
+
+	type testIn struct {
+		metricType        MetricsType
+		influxEnabled     bool
+		prometheusEnabled bool
+	}
+	type testOut struct {
+		expectedInfluxEnabled     bool
+		expectedprometheusEnabled bool
+	}
+
+	// test cases
+	type aTest struct {
+		description string
+		in          testIn
+		out         testOut
+	}
+	testCases := []aTest{
+		{
+			description: "[1] metricType = \"none\"; No flags enabled. ",
+			in:          testIn{"none", false, false},
+			out:         testOut{false, false},
+		},
+		{
+			description: "[2] metricType = \"none\"; Influx flag enabled.",
+			in:          testIn{"none", true, false},
+			out:         testOut{true, false},
+		},
+		{
+			description: "[3] metricType = \"none\"; Prometheus flag enabled. ",
+			in:          testIn{"none", false, true},
+			out:         testOut{false, true},
+		},
+		{
+			description: "[4] metricType = \"none\"; Both flags enabled. ",
+			in:          testIn{"none", true, true},
+			out:         testOut{true, true},
+		},
+		{
+			description: "[5] metricType = \"influx\"; No flags enabled.",
+			in:          testIn{"influx", false, false},
+			out:         testOut{true, false},
+		},
+		{
+			description: "[6] metricType = \"influx\"; Influx flag enabled.",
+			in:          testIn{"influx", true, false},
+			out:         testOut{true, false},
+		},
+		{
+			description: "[7] metricType = \"influx\"; Prometheus flag enabled.",
+			in:          testIn{"influx", false, true},
+			out:         testOut{true, true},
+		},
+		{
+			description: "[8] metricType = \"influx\"; Both flags enabled.",
+			in:          testIn{"influx", true, true},
+			out:         testOut{true, true},
+		},
+		{
+			description: "[9] metricType = \"prometheus\"; No flags enabled.",
+			in:          testIn{"prometheus", false, false},
+			out:         testOut{false, true},
+		},
+		{
+			description: "[10] metricType = \"prometheus\"; Influx flag enabled.",
+			in:          testIn{"prometheus", true, false},
+			out:         testOut{true, true},
+		},
+		{
+			description: "[11] metricType = \"prometheus\"; Prometheus flag enabled.",
+			in:          testIn{"prometheus", false, true},
+			out:         testOut{false, true},
+		},
+		{
+			description: "[12] metricType = \"prometheus\"; Both flags enabled.",
+			in:          testIn{"prometheus", true, true},
+			out:         testOut{true, true},
+		},
+		{
+			description: "[13] metricType = \"trendalyze\"; No flags enabled. ",
+			in:          testIn{"trendalyze", false, false},
```

Comment:
Consider renaming `trendalyze` to `unknown` for clarity in understanding the test values.

Reply:
Done